tests/sqlalchemy/test_datatype_parse.py

```diff
@@ -0,0 +1,180 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+from assertpy import assert_that
+from sqlalchemy.sql.sqltypes import (
+    CHAR, VARCHAR,
+    ARRAY,
+    INTEGER, DECIMAL,
+    DATE, TIME, TIMESTAMP
+)
+from sqlalchemy.sql.type_api import TypeEngine
+
+from trino.sqlalchemy import datatype
+from trino.sqlalchemy.datatype import MAP, ROW
+
+
+@pytest.mark.parametrize(
+    'type_str, sql_type',
+    datatype._type_map.items(),
+    ids=datatype._type_map.keys()
+)
+def test_parse_simple_type(type_str: str, sql_type: TypeEngine):
+    actual_type = datatype.parse_sqltype(type_str)
+    if not isinstance(actual_type, type):
+        actual_type = type(actual_type)
+    assert_that(actual_type).is_equal_to(sql_type)
+
+
+parse_cases_testcases = {
+    'char(10)': CHAR(10),
+    'Char(10)': CHAR(10),
+    'VARCHAR(10)': VARCHAR(10),
+    'varCHAR(10)': VARCHAR(10),
+    'VARchar(10)': VARCHAR(10),
```

Comment:
Can we test unbounded types too? `CHAR` and `VARCHAR`.

Reply:
Added test for unbounded types